src/l1block.rs

```diff
@@ -248,23 +248,123 @@ impl L1BlockInfo {
             .wrapping_div(TX_L1_FEE_PRECISION_U256) // account for penalty
     }
 
+    fn calculate_tx_l1_cost_galileo(
+        &self,
+        tx_size: u32, // size of the original rlp-encoded transaction
+        spec_id: ScrollSpecId,
+        compressed_size: u32, // size of the compressed rlp-encoded transaction
+    ) -> U256 {
+        // Post Galileo rollup fee formula:
+        // rollup_fee(tx) = fee_per_byte * compressed_size(tx) * (1 + penalty(tx)) / PRECISION
+        //
+        // Where:
+        // fee_per_byte = (exec_scalar * l1_base_fee + blob_scalar * l1_blob_base_fee)
+        // compressed_size(tx) = min(len(zstd(rlp(tx))), len(rlp(tx)))
+        // penalty(tx) = compressed_size(tx) / penalty_factor
+
+        assert!(
+            compressed_size <= tx_size,
+            "transaction compressed size {compressed_size} must be less than or equal to the original size {tx_size}"
+        );
```

Comment:
<img width="1520" height="286" alt="Image" src="https://github.com/user-attachments/assets/015a2b50-a618-45b2-aa5e-d04d422a86b9" />

it's not same as here, it will return original size if geth find the compressed_size > original size.

Reply:
agree this seems to be different behavior in l2geth